qiskit/algorithms/optimizers/steppable_optimizer.py

```diff
@@ -0,0 +1,252 @@
+from abc import abstractmethod, ABC
+from dataclasses import dataclass, field
+from typing import Dict, Any, Union, Callable, Optional, Tuple, List
+
+from .optimizer import Optimizer, POINT, OptimizerResult
+import numpy as np
+
+CALLBACK = Callable[[int, np.ndarray, float, float], None]
+
+
+@dataclass
+class AskObject(ABC):
+    """
+    Base class for return type of method :meth:`~qiskit.algorithms.optimizers.SteppableOptimizer.ask`.
+    Args:
+        x_fun: Point or list of points where the function needs to be evaluated to compute the
+            next state of the optimizer.
+        x_jac: Point or list of points where the gradient/jacobian needs to be evaluated to compute the
+            next state of the optimizer.
+    """
+
+    x_fun: Union[POINT, List[POINT], None] = None
```

Comment:
Typically we use `Optional[...]` instead of `Union[..., None]` (which is the exact purpose of `Optional`)
```suggestion
    x_fun: Optional[Union[POINT, List[POINT]]] = None
```

Reply:
Okay, I thought it was only when there were only 2 options. Good to know!